src/components/Modals/Send/views/Confirm.tsx

```diff
@@ -111,6 +121,111 @@ export const Confirm = ({ handleSubmit }: ConfirmProps) => {
     state: { wallet },
   } = useWallet()
 
+  const addressBookEntryFilter = useMemo(
+    () => ({ accountAddress: to, chainId: asset?.chainId }),
+    [to, asset?.chainId],
+  )
+  const addressBookEntry = useAppSelector(state =>
+    selectExternalAddressBookEntryByAddress(state, addressBookEntryFilter),
+  )
+
+  const internalAccountIdFilter = useMemo(
+    () => ({
+      accountAddress: to,
+      chainId: asset?.chainId,
+    }),
+    [to, asset?.chainId],
+  )
+
+  const internalAccountId = useAppSelector(state =>
+    selectInternalAccountIdByAddress(state, internalAccountIdFilter),
+  )
+
+  const accountMetadata = useAppSelector(selectPortfolioAccountMetadata)
+
+  const accountNumber = useMemo(
+    () =>
+      internalAccountId
+        ? accountMetadata[internalAccountId]?.bip44Params?.accountNumber
+        : undefined,
+    [accountMetadata, internalAccountId],
+  )
+
+  const internalAccountLabel = useMemo(() => {
+    if (!internalAccountId) return null
+
+    if (isUtxoAccountId(internalAccountId)) {
+      return accountIdToLabel(internalAccountId)
+    }
+
+    // Fallback to "Account" if accountNumber is not available yet
```

Comment:
q: What do we mean by "not available yet" ?

Reply:
Removed it and early returned, will see at runtime if it changes anything